pkg/operator/endpoint_status_builder.go

```diff
@@ -68,24 +68,116 @@ func (b *scrapeEndpointBuilder) add(target *prometheusv1.TargetsResult) error {
 	return nil
 }
 
+// convertScrapeKey converts the key to a CRD. See monitoringv1.PodMonitoringCRD.GetKey().
+func convertScrapeKey(key string) (monitoringv1.PodMonitoringCRD, error) {
+	split := strings.Split(key, "/")
+	// Generally:
+	// - 1 for scrape pools without a respective CRDs.
+	// - 2 for cluster-scoped resources.
+	// - 3 for namespaced resources.
+	switch split[0] {
+	case "kubelet":
+		if len(split) != 1 {
+			return nil, fmt.Errorf("invalid kubelet scrape key format %q", key)
+		}
+		return nil, nil
+	case "PodMonitoring":
+		if len(split) != 3 {
+			return nil, fmt.Errorf("invalid PodMonitoring scrape key format %q", key)
+		}
+		return &monitoringv1.PodMonitoring{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: split[1],
+				Name:      split[2],
+			},
+		}, nil
+	case "ClusterPodMonitoring":
+		if len(split) != 2 {
+			return nil, fmt.Errorf("invalid ClusterPodMonitoring scrape key format %q", key)
+		}
+		return &monitoringv1.ClusterPodMonitoring{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: split[1],
+			},
+		}, nil
+	case "NodeMonitoring":
+		if len(split) != 2 {
+			return nil, fmt.Errorf("invalid NodeMonitoring scrape key format %q", key)
+		}
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("unknown scrape kind %q", split[0])
+	}
+}
+
+// scrapePool is the parsed Prometheus scrape pool, which is assigned to the job name from our
+// configurations. For example, for PodMonitoring this is `PodMonitoring/namespace/name/port`. The
+// key is what idenitifies the resource (`PodMonitoring/namespace/name`) and the group indiciates a
+// small subset of that resource (`port`).
+type scrapePool struct {
+	key   string
+	group string
```

Comment:
nit (non-blocking): it may be more clear to just call the fields what you're describing in the comment, i.e. renaming `group` to `port`. Unless you think it's generalizable to another concept in our CRs...

Reply:
It's different in each CRD:
- PodMonitoring: port
- NodeMonitoring: path
- Probe: URL/index

Reply:
Ah touche - LGTM!